$NetBSD: patch-configure.ac,v 1.4 2021/09/30 10:32:20 jperkin Exp $

Don't use non-portable operator ==:
https://gitlab.haskell.org/ghc/ghc/merge_requests/2497

Copy files rather than relying on hardlink behaviour.

--- configure.ac.orig	2020-07-12 16:04:30.000000000 +0000
+++ configure.ac
@@ -685,11 +685,11 @@ dnl ------------------------------------
 dnl ** Copy the files from the "fs" utility into the right folders.
 dnl --------------------------------------------------------------
 AC_MSG_NOTICE([Creating links for in-tree file handling routines.])
-ln -f utils/fs/fs.* utils/lndir/
-ln -f utils/fs/fs.* utils/unlit/
-ln -f utils/fs/fs.* rts/
-ln -f utils/fs/fs.h libraries/base/include/
-ln -f utils/fs/fs.c libraries/base/cbits/
+cp -p utils/fs/fs.* utils/lndir/
+cp -p utils/fs/fs.* utils/unlit/
+cp -p utils/fs/fs.* rts/
+cp -p utils/fs/fs.h libraries/base/include/
+cp -p utils/fs/fs.c libraries/base/cbits/
 AC_MSG_NOTICE([Routines in place. Packages can now be build normally.])
 
 dnl --------------------------------------------------------------
@@ -843,7 +843,7 @@ if test "x$EnableDtrace" = "xyes"; then
   if test -n "$DtraceCmd"; then
     if test "x$TargetOS_CPP-$TargetVendor_CPP" = "xdarwin-apple" \
       -o "x$TargetOS_CPP-$TargetVendor_CPP" = "xfreebsd-portbld" \
-      -o "x$TargetOS_CPP-$TargetVendor_CPP" = "xsolaris2-unknown"; then
+      ; then
       HaveDtrace=YES
     fi
   fi
@@ -950,7 +950,7 @@ FP_CHECK_SIZEOF_AND_ALIGNMENT(uint64_t)
 
 dnl for use in settings.in
 TargetWordSize=$ac_cv_sizeof_void_p
-if test "x$TargetWordSize" == 8; then
+if test "x$TargetWordSize" = 8; then
   AC_SUBST([Cabal64bit],[True])
 else
   AC_SUBST([Cabal64bit],[False])
